Theming updates to the Recruiter Dashboard, Modal window and Checkout pages

BenPoole88 opened the pull request:
Hey Rob, I'm so sorry I haven't been able to completely finish the theming for the recruiter journey this week, my issues with the CJ block this morning ate up a lot of my time. I've pushed everything I've done so far, if you could review what I've done for Monday next week, we can go from there!

Commit:
further theming updates to job post form

Commit:
Merge branch 'jobposttheme' of github.com:BenPoole88/christianjobs into jobposttheme

Commit:
Merge branch '2.0.0-dev' into jobposttheme

Commit:
initial theming for recruiter pages

Commit:
Merge branch '2.0.0-dev' of github.com:rlmumford/christianjobs into recruiterpagestheme

Commit:
Merge branch '2.0.0-dev' of github.com:rlmumford/christianjobs into recruiterpagestheme

Commit:
Merge branch '2.0.0-dev' of github.com:rlmumford/christianjobs into recruiterpagestheme

Commit:
font additions to scss file

Commit:
summary label display set to 'none' in forms only

Commit:
Merge branch '2.0.0-dev' of github.com:rlmumford/christianjobs into recruiterpagestheme

Commit:
further theming updated to recruiter journey

Commit:
Merge branch '2.0.0-dev' into recruiterpagestheme

Commit:
further theming updates to recruiter journey

Commit:
further updates to recruiter theming

Commit:
further updates to recruiter theming

Commit:
further theming for recruiter journey

Commit:
theming updates to the recruiter dashboard, modal window and checkout pages

BenPoole88 commented:
As per your request, I've added everything into this one PR. The styling updates to the recruiter journey currently include the styling of:

- the Recruiter Dashboard tabs
- the Recruiter View, Edit, Team, and Jobs Pages
- the popup modal window (though I'm having trouble getting this to appear currently)
- the Checkout process

![Recruiter Dashboard - View](https://user-images.githubusercontent.com/58986635/85390981-7cd42100-b541-11ea-924c-93c2f31297b5.png)

![Recruiter Dashboard Edit](https://user-images.githubusercontent.com/58986635/85390996-82ca0200-b541-11ea-8781-6570fb580fd4.png)

![Recruiter Dashboard Edit 2](https://user-images.githubusercontent.com/58986635/85391011-89587980-b541-11ea-81ff-d3f22ffe0a38.png)

![Recruiter Dashboard Jobs](https://user-images.githubusercontent.com/58986635/85391026-8f4e5a80-b541-11ea-9b5b-1fb247e3aa9f.png)

![Recruiter Dashboard Team](https://user-images.githubusercontent.com/58986635/85391044-95dcd200-b541-11ea-8572-208876456299.png)

![Checkout Order Information](https://user-images.githubusercontent.com/58986635/85391071-9ecda380-b541-11ea-89ce-7950f2af0e62.png)

![Checkout Review](https://user-images.githubusercontent.com/58986635/85391088-a4c38480-b541-11ea-9c4b-48c21599b8e3.png)

rlmumford commented:
No screenshot of the popup window there.
Second line of address is out of line.
I think generally on the checkout "Edit" and "Remove" need to look less eye catching than continue.
Can we bring the down arrow on "Approve" into the button itself (and a screenshot of when you click the down arrow).

BenPoole88 commented:
I've updated the Remove button, so it sits in the background a little more, see below:

![Recruiter Remove Button](https://user-images.githubusercontent.com/58986635/85395609-51086980-b548-11ea-9699-5e5842b52ced.png)

And here's the dropdown for the Accept/Remove buttons:

![Recruiter Team](https://user-images.githubusercontent.com/58986635/85395693-7006fb80-b548-11ea-92ea-9f6d971a394d.png)

The other issues you mentioned, I could do with chatting through with you!

BenPoole88 committed:
further styling updates to recruiter journey

BenPoole88 committed:
Merge branch '2.0.0-dev' into recruiterpagestheme

BenPoole88 commented:
Here's a shot of the popup window:

![Modal](https://user-images.githubusercontent.com/58986635/85413773-24ae1680-b563-11ea-8bc1-3e993e498a3b.png)

BenPoole88 committed:
further theming updates to recruiter dashboard and checkout flow

BenPoole88 commented:
This afternoon I've actioned the bits we discussed, including fixing the text box alignment:

![Text Box Alignment Fix](https://user-images.githubusercontent.com/58986635/85427677-fc7ae380-b573-11ea-8f11-115524437af9.png)

Styling the Accept/Reject dropdown:

![Approve](https://user-images.githubusercontent.com/58986635/85427726-11577700-b574-11ea-9ce7-2cb884bd82cc.png)
![Reject](https://user-images.githubusercontent.com/58986635/85427737-15839480-b574-11ea-8425-f844e5f57492.png)

Styled the checkout complete box:

![Checkout Complete Box](https://user-images.githubusercontent.com/58986635/85427978-6abfa600-b574-11ea-8bdf-f92839b65abc.png)

And started a bit of initial work on the New Organization button:

![New Organization](https://user-images.githubusercontent.com/58986635/85428032-7e6b0c80-b574-11ea-82ca-be1c71633ad3.png)

rlmumford commented on web/modules/custom/job_board/src/Plugin/Block/RecruiterMenu.php:
```diff
@@ -96,8 +96,8 @@ public function build() {
     if ($user->organization->isEmpty()) {
-      $cache->applyTo($build);
-      return $build;
+      //$cache->applyTo($build);
+      //return $build;
```
We need to not commit these two lines.

BenPoole88 replied:
Of course, my bad! I've removed the forward slashes from both lines now and will push through.

rlmumford commented on web/themes/custom/fmcg_material/sass/_button.scss:
```diff
@@ -30,3 +30,66 @@ input.button, .btn, a.btn, span.btn, .tabledrag-toggle-weight, input[type="submi
+  list-style-image: none;
+}
+
+li.approve, li.reject {
```
I'm just concerned about how this will work if we add a third button to the dropbutton (or even just rename one to a different name). It looks like the styles won't apply beyond the approve and reject button.

replacing this with 
```css
.dropbutton li {
```

rlmumford commented on web/themes/custom/fmcg_material/sass/_button.scss:
```diff
@@ -30,3 +30,66 @@ input.button, .btn, a.btn, span.btn, .tabledrag-toggle-weight, input[type="submi
+  }
+}
+
+.js .dropbutton-widget .dropbutton {
```
Might not need the .js class

rlmumford commented on web/themes/custom/fmcg_material/sass/_button.scss:
```diff
@@ -30,3 +30,66 @@ input.button, .btn, a.btn, span.btn, .tabledrag-toggle-weight, input[type="submi
+  }
+}
+
+li.approve {
+  margin: 8px;
+}
+
+li.reject {
+  margin: -8px 0px 0px 8px;
+}
```
Work out how to to do these without using approve/reject. It might be about :first-child and :last-child?

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -1913,39 +1984,78 @@ form#views-exposed-form-job-board-page {
+    font-size: 30px;
+    font-weight: 600;
+    margin-bottom: -24px;
+    color: #67C367;
```
Make this colour a variable.

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -2192,3 +2308,36 @@ form#views-exposed-form-job-board-page {
+  }
+}
+
+.field-name-name {
```
A bit concerned about this selector? What is it doing?

BenPoole88 replied:
This is the field that contains the organization name, that appears above the recruiter dashboard tabs.

rlmumford replied:
Can we wrap it in a class associated with the block then? Just don't want this to apply to every field called "name" anywhere in the website.

rlmumford replied:
@BenPoole88 this doesn't seem to have been done?

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -2192,3 +2308,36 @@ form#views-exposed-form-job-board-page {
+  }
+}
+
+.form-type-radio {
+  label.option {
+    width: 100%;
+    display: block;
+  }
+}
```
Shouldn't this go in the _form.scss file?

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -1891,13 +1957,18 @@ form#views-exposed-form-job-board-page {
     padding-left: 25px;
   }
 
+  td, td:first-child, td:last-child {
```
Are the first- and last- child needed here? They might be to override other styles.

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -1844,16 +1839,89 @@ form#views-exposed-form-job-board-page {
+    }
+  }
+
+  .col-md-8, .col-xs-12 {
```
Is there a way of dealing with this without altering the styles of the .col-* selectors?

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -1844,16 +1839,89 @@ form#views-exposed-form-job-board-page {
+    background-color: $fmcg-black;
+
+    &::after {
+        content: "";
```
Too much identation.

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -1844,16 +1839,89 @@ form#views-exposed-form-job-board-page {
 //Checkout Pages
 
 .layout-checkout-form {
```
Maybe go into a __checkout.scss file?

rlmumford commented on web/themes/custom/fmcg_material/sass/styles.scss:
```diff
@@ -1844,16 +1839,89 @@ form#views-exposed-form-job-board-page {
+  div.form-item label {
+    font-size: 14px;
+    font-weight: 400;
+    font-family: Poppins;
```
I'm surprised we need to set the font family and weight here.

rlmumford commented:
Really like how things are looking here. There are quite a few code style changes specific to the implementation detail.

I like the drop button going chrome - but lets also make the "Change Role" button go chrome as well. It might be that we use a selector like `.operations .btn` or whatever it is.

Popup looks good. I don't love the styling of the close button, and it could possibly do with a bit more box shadow to make it feel "on top" of the page.

BenPoole88 committed:
further theming updates and amends to recruiter journey

BenPoole88 committed:
Merge branch '2.0.0-dev' into recruiterpagestheme

BenPoole88 committed:
further theming and code amends to PR #23

BenPoole88 commented:
I've carried out all the above requested code changes, I think! I've also added a subtle hover to the drop button:

![Chrome Dropbutton with Hover](https://user-images.githubusercontent.com/58986635/85547333-b4140200-b615-11ea-9051-2e626275fcc6.png)

rlmumford commented:
I think it would be good to leave any "Recruiter Menu" changes out of this PR - as that seems to be a fair ways off at the moment and is a fairly distinct piece of work..

rlmumford commented on web/themes/custom/fmcg_material/sass/_button.scss:
```diff
@@ -55,20 +55,30 @@ input.button, .btn, a.btn, span.btn, .tabledrag-toggle-weight, .button-action, .
+    border: 2px solid $fmcg-chrome;
+  }
+
   a {
```
So this is now going to apply to any a tag inside an a tag inside something with the class drop button.

I think that means its not actually going to apply to anything, so we can probably remove

```html
<div class="dropbutton">
  <a>
    <a></a><!-- Will only apply to this internal a tag -->
  </a>
</div>
```

BenPoole88 replied:
Ah I understand this better now, this diagram helps me to better see the hierarchy.

BenPoole88 commented:
> I think it would be good to leave any "Recruiter Menu" changes out of this PR - as that seems to be a fair ways off at the moment and is a fairly distinct piece of work..

Yes, agreed. I think once I we get to it, it'll have its own branch and PR.

rlmumford commented on web/themes/custom/fmcg_material/sass/_checkout.scss:
```diff
@@ -0,0 +1,220 @@
+  }
+}
+
+.layout-region {
```
Is this true for all layouts everywhere, or just when they're inside the checkout?

BenPoole88 replied:
This is specifically for the checkout, so I've now wrapped this in the appropriate class.

rlmumford commented on web/themes/custom/fmcg_material/sass/_form.scss:
```diff
@@ -11,19 +11,26 @@ form {
     word-wrap: normal;
   }
 
+  .field--name-description-summary .filter-wrapper {
```
Is this only desired for the description summary field? OR for all formatted text fields?

BenPoole88 replied:
As discussed, after not being able to find where this rule is in effect, I've deleted it.

rlmumford reviewed (changes requested):
A few more things still to consider.

BenPoole88 committed:
further code amends as recorded in PR #23

BenPoole88 commented:
So I've been having a look at the table headers, as we discussed. I've added in a background image as well, just for the purpose of being able to see the form. I'm not married to any of them really, they're more to show you roughly where my head is with it, I do agree with you though it's far less intense when the header isn't fmcg-black so I think that's a good starting point. See what you think!

![25 6 20 Table V1](https://user-images.githubusercontent.com/58986635/85699624-97d69a80-b6d3-11ea-9dd3-37baedb37b78.png)

![25 6 20 Table V2](https://user-images.githubusercontent.com/58986635/85699658-9efda880-b6d3-11ea-9593-1025d39027ed.png)

On this one, I've also added in a slightly different grey:
![25 6 20 Table V3](https://user-images.githubusercontent.com/58986635/85699684-a3c25c80-b6d3-11ea-9de6-f30563d6eea3.png)

rlmumford commented:
As I said on hangouts, If we're going to box the page content, I think we need to think about here the tags go, how the recruiter menu looks and everything.

Of the boxed pages, the first one is my favourite. I don't know if we could add a very subtle transparency to the white background or make it a slight grey. Just trying to take some of the boldness away.

BenPoole88 committed:
further styling to recruiter journey PR #23

BenPoole88 commented:
I think a slight transparency may work yes, see below. I think the background image we'll need to choose very carefully though:

![25 6 20 Table V4](https://user-images.githubusercontent.com/58986635/85751591-9e303b00-b702-11ea-9825-75dbe7a87ec0.png)

rlmumford commented:
Yes. That's the first box that I like.

I think make the tabs bar touch the edges & take the logo and title out of the box. 

Not sure what to do about the recruiter menu. Maybe put it somewhere else entirely? A single icon that you click to expand?

BenPoole88 committed:
theme updates to table headers and main content background

BenPoole88 commented:
I've done an update to the table headers:

![Screenshot 2020-06-29 at 12 48 12 (2)](https://user-images.githubusercontent.com/58986635/86001638-151d4a80-ba07-11ea-8a3b-fbf904cf1a79.png)

BenPoole88 committed:
removing padding on recruiter tabs

BenPoole88 commented:
Updated with the recruiter tabs padding removed:

![Screenshot 2020-06-29 at 13 05 31 (2)](https://user-images.githubusercontent.com/58986635/86003116-3e3eda80-ba09-11ea-9383-6bb192847708.png)

BenPoole88 committed:
nested h2 inside recruiter dashboard header
